src/beat_number.rs

```diff
@@ -7,4 +8,83 @@ impl BeatNumber {
     pub fn is_beginning_of_measure(&self) -> bool {
         self.quarter_note == 1 && self.sixteenth_note == 1
     }
+
+    pub fn minus_sixteenths(&self, num_sixteenths: u32) -> BeatNumber {
+        if num_sixteenths < self.sixteenth_note {
+            return BeatNumber {
+                quarter_note: self.quarter_note,
+                sixteenth_note: self.sixteenth_note - num_sixteenths,
+            };
+        }
+        let num_quarter_notes = (num_sixteenths + 3) / 4;
+        BeatNumber {
+            quarter_note: (self.quarter_note as i32 - num_quarter_notes as i32 - 1) as u32 % 4 + 1,
+            sixteenth_note: (self.sixteenth_note as i32 - num_sixteenths as i32 - 1) as u32 % 4 + 1,
+        }
+    }
```

Comment:
Is there a benefit to having separate `quarter_note` and `sixteenth_note` fields on this struct? It seems like that is baking in something that could maybe just be exposed as convenience functions. You could store a single number `sixteenth_note` which represents the sixteenth note of the entire bar the `BeatNumber` represents, and then have `quarter_note` and `sixteenth_note_in_beat` functions which expose the numbers currently being saved in these fields.